testing/common_testing_util.py

```diff
@@ -0,0 +1,110 @@
+from pathlib import Path
+import pytest
+from ivadomed.scripts import download_data as ivadomed_download_data
+import os
+import shutil
+import sys
+
+# Cannot use __ivadomed_dir__ here as that would refer to the IvadoMed installed in the Conda/Venv
+# Absolute path to the temporary folder, which should sit at the root of the repo.
+# We could also replace this hard coded levels reference with a simple loop to search for .git/.github sitting at the
+# root of the repo. To be discussed.
+FOLDER_LEVELS_FROM_ROOT = 1
+
+path_repo_root: Path = Path(__file__).parents[FOLDER_LEVELS_FROM_ROOT].absolute()
+
+path_temp: str = str(path_repo_root / "tmp")
+
+path_unit_tests: str = str(path_repo_root / "testing" / "unit_tests")
+path_data_testing_source: str = str(path_repo_root / "data_testing")
+path_data_testing_tmp: str = str(path_repo_root / "tmp" / Path(path_data_testing_source).name)
+
+path_functional_tests: str = str(path_repo_root / "testing" / "functional_tests")
+path_data_functional_source: str = str(path_repo_root / "data_functional_testing")
+path_data_functional_tmp: str = str(path_repo_root / "tmp" / Path(path_data_functional_source).name)
```

Comment:
These are probably the most important changes. 
These path variables are used to supplement, covertly replace the older __temp__ etc variables in the individual t_util.pys that are imported by either /unit_tests or /functional_test

Next PR update will then address those in the individual files given there are potential ongoing PR implicating those tests.